consensus/src/txn_cache.rs

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_crypto::ed25519::{Ed25519PrivateKey, Ed25519Signature};
+use aptos_crypto::{hash::DefaultHasher, HashValue, PrivateKey, Uniform};
+use aptos_types::account_address::AccountAddress;
+use aptos_types::chain_id::ChainId;
+use aptos_types::transaction::{RawTransaction, Script, SignedTransaction};
+use bcs::to_bytes;
+use concurrent_lru::sharded::LruCache;
```

Comment:
[their repo](https://github.com/ngkv/concurrent_lru)
impl looks close to what we want, but either the cache API is weird (force me to use the `mysterious_counter` below) or I'm not good at reading rust.